### DISCUSSION TITLE
Propagating Theme to 3rd Party Controls

### DISCUSSION THREAD

**Discussion by mysteryx93** [AvaloniaUI/Avalonia#8632] [Q&A]
Got the Light/Dark themes working for my app. Except for the media player.

What's the best method of propagating the theme to a 3rd party control like the media player?
https://github.com/mysteryx93/MediaPlayerUI.NET/

I can put a Theme property on each individual control, but is that the best way? Or have a static property; or do I need to implement a ThemeManager within the library? I just don't want to re-invent the wheel here.

Probably a part of the solution would be to use standard brush resources for system colors; but that can vary depending what theme library is used. Is there a list of standard resources that I can expect?

I can have my resources in DarkTheme.axaml and replace that file with LightTheme.axaml, dynamically loading/unloading merged dictionaries. Then the best would be to have an internal ThemeManager that handles that?

**Comment by maxkatz6** [comment]
It's not possible with build-in avalonia features yet. See https://github.com/AvaloniaUI/Avalonia/discussions/7886 and https://github.com/AvaloniaUI/Avalonia/pull/8166. It's expected in the 11.0.
Long story short, instead of theme specific color mode, there will be one per application inherited on its controls. At the same time you can override color mode (or theme variant) on specific controls tree.

> I can put a Theme property on each individual control, but is that the best way? 

You can do it now, and I think it should be easy for users to update it later with 11.0 API.
The problem with 0.10 version that no control knows what color mode is set, as it might be defined differently in a different theme (material, fluent from this repo, fluent v2 theme from FluentAvalonia...),

**Reply by mysteryx93** [reply]
hum... so should I wait for 11.0 before adapting the class library? When is it expected to come out?

**Comment by kekekeks** [comment]
I think for this particular control you are expected to supply the colors manually since it doesn't use Avalonia's theme resources and defines its own https://github.com/mysteryx93/MediaPlayerUI.NET/blob/master/Avalonia/Styles/Colors/Gray.axaml

**Reply by mysteryx93** [reply]
It's my control. I'll redesign how it handles resources based on recommended standards -- once a standard is established.
